### PR TITLE
fix(gatsby-plugin-theme-ui,docs): Dependency Updates

### PR DESCRIPTION
**Pull request by LekoArts**
Hi!

This PR does a couple of things:

- Update the two Gatsby examples to use Gatsby v4 + add necessary peerDeps
- Update `gatsby-plugin-theme-ui` to add necessary peerDeps
- Update README of `gatsby-plugin-theme-ui` and docs to include installation of  peerDeps

# Version

Published prerelease version: `v0.14.0-develop.15`

<details>
  <summary>Changelog</summary>

  :tada: This release contains work from new contributors! :tada:
  
  Thanks for all your work!
  
  :heart: Brage ([@braaar](https://github.com/braaar))
  
  :heart: peterlits zo ([@PeterlitsZo](https://github.com/PeterlitsZo))
  
  :heart: Ryan Turner ([@rtturner](https://github.com/rtturner))
  
  :heart: Cannon Lock ([@CannonLock](https://github.com/CannonLock))
  
  #### 🚀 Enhancement
  
  - feat(examples/next): Add new deps, fully use TSX, rebuild [#2068](https://github.com/system-ui/theme-ui/pull/2068) ([@lachlanjc](https://github.com/lachlanjc))
  - `@theme-ui/tailwind`
    - feat(tailwind): Upgrade Tailwind theme conversion for v3.0 [#2082](https://github.com/system-ui/theme-ui/pull/2082) ([@lachlanjc](https://github.com/lachlanjc))
  - `@theme-ui/custom-properties`
    - feat(custom-properties): Warn in development on invalid theme keys [#2080](https://github.com/system-ui/theme-ui/pull/2080) ([@lachlanjc](https://github.com/lachlanjc))
  - `@theme-ui/color-modes`
    - feat(color-modes): Warn when theme color keys have leading/trailing whitespace [#2099](https://github.com/system-ui/theme-ui/pull/2099) ([@lachlanjc](https://github.com/lachlanjc))
  - `theme-ui`
    - feat(theme-ui): Add sx prop to BaseStyles component [#2081](https://github.com/system-ui/theme-ui/pull/2081) ([@lachlanjc](https://github.com/lachlanjc))
  
  #### 🐛 Bug Fix
  
  - feat(docs): Add custom content for sketchy preset demo [#1236](https://github.com/system-ui/theme-ui/pull/1236) ([@atanasster](https://github.com/atanasster) [@hasparus](https://github.com/hasparus) [@lachlanjc](https://github.com/lachlanjc))
  - Finish editing note [#2079](https://github.com/system-ui/theme-ui/pull/2079) ([@lachlanjc](https://github.com/lachlanjc))
  - fix(docs): Update recommendations in Keyframes doc [#2079](https://github.com/system-ui/theme-ui/pull/2079) ([@lachlanjc](https://github.com/lachlanjc))
  - Release v0.13.2 [#2075](https://github.com/system-ui/theme-ui/pull/2075) ([@lachlanjc](https://github.com/lachlanjc) [@hasparus](https://github.com/hasparus) [@braaar](https://github.com/braaar) [@dependabot[bot]](https://github.com/dependabot[bot]))
  - `gatsby-plugin-theme-ui`
    - fix(gatsby-plugin-theme-ui,docs): Dependency Updates [#2138](https://github.com/system-ui/theme-ui/pull/2138) ([@LekoArts](https://github.com/LekoArts))
  - `@theme-ui/style-guide`
    - fix(style-guide): use inherited fontSize for ColorPalette color labels [#2135](https://github.com/system-ui/theme-ui/pull/2135) ([@braaar](https://github.com/braaar))
  - `@theme-ui/components`
    - fix(components): Update IconButton type definition to include size prop [#2121](https://github.com/system-ui/theme-ui/pull/2121) ([@rtturner](https://github.com/rtturner))
    - components: Fix visual collapsing bug with Switch component [#2067](https://github.com/system-ui/theme-ui/pull/2067) ([@lachlanjc](https://github.com/lachlanjc))
  - `@theme-ui/sidenav`
    - fix(sidenav): Build with latest theme-ui version [#2084](https://github.com/system-ui/theme-ui/pull/2084) ([@lachlanjc](https://github.com/lachlanjc))
  
  #### 👨‍💻 Minor changes
  
  - Update jsx-pragma.mdx ([@hasparus](https://github.com/hasparus))
  - docs(examples/next): fix case insensitive import ([@hasparus](https://github.com/hasparus))
  
  #### 🏠 Internal
  
  - docs(sx-prop): add more detail for responsive values. [#2116](https://github.com/system-ui/theme-ui/pull/2116) ([@PeterlitsZo](https://github.com/PeterlitsZo) [@lachlanjc](https://github.com/lachlanjc))
  - docs(jsx-pragma): Add section on using with vite [#2119](https://github.com/system-ui/theme-ui/pull/2119) ([@PeterlitsZo](https://github.com/PeterlitsZo) [@lachlanjc](https://github.com/lachlanjc))
  - docs: Clarify usage of theme prop in MDX Layout docs [#2100](https://github.com/system-ui/theme-ui/pull/2100) ([@CannonLock](https://github.com/CannonLock))
  - docs(jsx): add some more detail to #with-swc section in jsx-pragma docs [#2094](https://github.com/system-ui/theme-ui/pull/2094) ([@hasparus](https://github.com/hasparus) [@lachlanjc](https://github.com/lachlanjc))
  - e2e: Upgrade dependencies [#2098](https://github.com/system-ui/theme-ui/pull/2098) ([@lachlanjc](https://github.com/lachlanjc))
  - Set up CodeSandbox CI [#2085](https://github.com/system-ui/theme-ui/pull/2085) ([@lachlanjc](https://github.com/lachlanjc))
  - examples: Remove CodeSandbox starter example [#2086](https://github.com/system-ui/theme-ui/pull/2086) ([@lachlanjc](https://github.com/lachlanjc))
  - docs: Document default theme values [#2087](https://github.com/system-ui/theme-ui/pull/2087) ([@lachlanjc](https://github.com/lachlanjc))
  - `@theme-ui/sidenav`
    - sidenav: Fix props leaking to DOM on Pagination component [#2057](https://github.com/system-ui/theme-ui/pull/2057) ([@lachlanjc](https://github.com/lachlanjc))
  
  #### Authors: 9
  
  - [@dependabot[bot]](https://github.com/dependabot[bot])
  - Atanas Stoyanov ([@atanasster](https://github.com/atanasster))
  - Brage ([@braaar](https://github.com/braaar))
  - Cannon Lock ([@CannonLock](https://github.com/CannonLock))
  - Lachlan Campbell ([@lachlanjc](https://github.com/lachlanjc))
  - Lennart ([@LekoArts](https://github.com/LekoArts))
  - peterlits zo ([@PeterlitsZo](https://github.com/PeterlitsZo))
  - Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))
  - Ryan Turner ([@rtturner](https://github.com/rtturner))
</details>

### CODE REVIEW

File: examples/gatsby-plugin/package.json [hunk 1]
```diff
@@ -6,25 +6,21 @@
...
+    "gatsby": "^4.7.2",
+    "gatsby-plugin-mdx": "^3.7.1",
     "gatsby-plugin-theme-ui": "latest",
-    "react": "^16.14.0",
-    "react-dom": "^16.14.0",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
     "theme-ui": "latest"
-  },
-  "devDependencies": {
```
Comment (LekoArts) [inline examples/gatsby-plugin/package.json:23]: Those seemed unused

File: examples/gatsby/src/layout.tsx [hunk 2]
```diff
@@ -1,11 +1,8 @@
 /** @jsx jsx */
 import { FC } from 'react'
-import { jsx, Box, Container } from 'theme-ui'
-import { useBreakpointIndex } from '@theme-ui/match-media'
+import { Box, Container } from 'theme-ui'
 
 export const Layout: FC = ({ children }) => {
-  const index = useBreakpointIndex({ defaultIndex: 2 })
```
Comment (LekoArts) [inline examples/gatsby/src/layout.tsx:7]: Was not used anywhere

File: packages/docs/src/pages/getting-started/gatsby.mdx [hunk 3]
```diff
@@ -7,7 +7,7 @@ title: Getting Started with Gatsby
 To use Theme UI with [Gatsby][], install and use [`gatsby-plugin-theme-ui`](/packages/gatsby-plugin).
 
 ```sh
-npm i theme-ui gatsby-plugin-theme-ui
+npm i theme-ui gatsby-plugin-theme-ui @emotion/react @emotion/styled @mdx-js/react@v1
```
Comment (LekoArts) [inline packages/docs/src/pages/getting-started/gatsby.mdx:10]: `gatsby-plugin-mdx` doesn't support ESM-only MDX v2 yet, so people need to install v1

File: examples/gatsby-plugin/src/gatsby-plugin-theme-ui/index.js [hunk 4]
```diff
@@ -1,6 +1,5 @@
 const theme = {
-  useCustomProperties: true,
-  initialColorMode: 'light',
+  initialColorModeName: 'light',
```
Comment (lachlanjc) [inline examples/gatsby-plugin/src/gatsby-plugin-theme-ui/index.js]:
```suggestion
  config: {
    initialColorModeName: 'light',
  },
```

File: examples/gatsby/src/theme.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
-import { Theme } from 'theme-ui'
+import type { Theme } from 'theme-ui'
 
 const theme: Theme = {
+  initialColorModeName: 'light',
```
Comment (lachlanjc) [inline examples/gatsby/src/theme.ts]:
```suggestion
  config: {
    initialColorModeName: 'light',
  },
```

**Review by lachlanjc — APPROVED** [review]
Thank you so much!

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.14.0-develop.15`](https://github.com/system-ui/theme-ui/releases/tag/v0.14.0-develop.15) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.14.0`](https://github.com/system-ui/theme-ui/releases/tag/v0.14.0) :rocket:

**Pull request merged**